source/com.microsoft.tfs.core/src/com/microsoft/tfs/core/telemetry/TfsTelemetryHelper.java

```diff
@@ -75,22 +81,42 @@ public synchronized static TelemetryClient getTelemetryClient() {
     }
 
     public static void sendMetric(final String name, final double value) {
+        if (sendingTelemetryDisabled) {
```

Comment:
I agree no-op clients are the best way to be safe, this would shield us from bugs in TelemetryClient as well.  Just want to point out since TelemetryClient isn't an interface, it would expose (in the future) new public classes and our dummy may suddenly gain new abilities.  However if all our access to this client is again encapsulated in our own helpers, the risk is pretty low IMHO.